src/PerforceCommands.ts

```diff
@@ -152,25 +152,14 @@ export namespace PerforceCommands {
             return false;
         }
 
-        //If folder not opened, overrided p4 directory
         const fileUri = editor.document.uri;
-        const directoryOverride = !checkFolderOpened()
-            ? Path.dirname(fileUri.fsPath)
-            : undefined;
+        p4revert(fileUri);
+    }
 
-        const args = [Utils.expansePath(fileUri.fsPath)];
-        PerforceService.execute(
-            fileUri,
-            "revert",
-            (err, stdout, stderr) => {
-                PerforceService.handleCommonServiceResponse(err, stdout, stderr);
-                if (!err && !stderr) {
-                    Display.showMessage("file reverted");
-                }
-            },
-            args,
-            directoryOverride
-        );
+    export async function p4revert(fileUri: Uri) {
+        const revertOpts: p4.RevertOptions = { paths: [fileUri] };
+        await p4.revert(fileUri, revertOpts);
+        Display.showMessage(fileUri.fsPath + " reverted.");
     }
```

Comment:
just occurred to me, that this won't automatically refresh the scm provider view or the status item - previously this was done by `handleCommonServiceResponse` which is no longer called

(All the perforce api functions *will* still show errors in the output / status bar, but it doesn't make any assumptions about whether you want to refresh the view, since that is a relatively expensive / obtrusive operation, so we only need the refreshing bits here - This may be a good use case for [decorators](https://www.typescriptlang.org/docs/handbook/decorators.html) but that's probably a wider job so would just do the simple way for now)

Reply:
I resolved this by updating the scm provider in the event calls (either when adding or deleting a file) as this is at a higher level and can be done once per action on multiple files rather than every time a revert or delete is done.